### PR TITLE
feat(fixpnt): add fused fma(a,b,c) with single rounding

### LINKED ISSUE #1191: feat(fixpnt): add fused fma(a,b,c)
Sub-issue of #1189 (universal fma).

`fixpnt` has no `fma`. Add a free `fma(a,b,c)` in `sw::universal` that forms `a*b + c` in a wide intermediate before the single fixpnt rounding/overflow step (fused: one rounding).

## Acceptance
- [ ] `sw::universal::fma` for `fixpnt`, `std::fma`-compatible signature.
- [ ] Regression test (single-rounding vs wide-oracle, saturation/overflow cases).
- [ ] Green on gcc + clang.

Parent: #1189.

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary
Sub-issue of the universal fma epic #1189. `fixpnt` had no `fma`; this adds a free `sw::universal::fma(a,b,c)` that forms `a*b + c` in a wide exact intermediate and rounds **once** (fused), matching `std::fma`'s contract.

- The product `a*b` is formed **exactly** as a `2*nbits` two's-complement value via `urmul2` (the same primitive `fixpnt::operator*=` uses).
- `c` is sign-extended and shifted into the product's `2*rbits`-fractional frame and added **exactly**.
- Only then is the result rounded (round-to-nearest-even at `rbits`, via `blockbinary::roundingMode`) and range-limited (**Modulo** wrap / **Saturate** clamp), mirroring `operator*=`'s rounding model but with `c` folded in before the single rounding.

So `fma` is strictly more accurate than the two-rounding `fixpnt(a*b) + c` whenever the product's low-order bits matter (cancellation).

## Changes
- `include/sw/universal/number/fixpnt/math/fma.hpp` -- new free `fma(a,b,c)`.
- `include/sw/universal/number/fixpnt/mathlib.hpp` -- include it.
- `include/sw/universal/number/fixpnt/fixpnt_fwd.hpp` -- forward declaration.
- `static/fixpnt/binary/math/fma.cpp` -- regression suite (target `fixpnt_fma`).

## Verification -- independent oracle
The test validates against an **independent `__int128` integer oracle**: `a,b,c` are exact multiples of `2^-rbits`, so `a*b+c` is an exact rational `(A*B + C*2^rbits)/2^(2*rbits)`; the oracle rounds it round-to-nearest-even at `rbits` then applies the range policy -- entirely separate from the implementation.

- exhaustive `(a,b)` x fixed-`c` for `fixpnt<8,4>` (Modulo + Saturate) and `fixpnt<8,0>`
- randomized correctly-rounded vs oracle: `fixpnt<16,8>`, `<24,12>`, `<32,16>`, both policies
- **single-rounding superiority**: the naive two-rounding `a*b + c` disagreed with the exact result in **30 / 5000** cases; `fma` matched the oracle in **all** of them
- identities (`fma(a,b,0)==a*b`, `fma(0,b,c)==c`, `fma(1,b,c)==b+c`) and saturation/overflow corners

## Test Results
| Target | gcc build | gcc test | clang build | clang test |
|--------|-----------|----------|-------------|------------|
| fixpnt_fma | OK | PASS | OK | PASS |

## Note
The wide accumulator must be **explicitly zeroed** -- `blockbinary` is trivially constructible, so gcc happened to clear the stack but clang did not, which leaked into `fma(0,b,c)`. Caught and fixed before the clang run (the documented gcc/clang stack-init portability gotcha).

## Test plan
- [ ] Fast CI passes (gcc + clang CI_LITE)
- [ ] Promote to ready when satisfied: `gh pr ready`

Resolves #1191
Relates to #1189

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added fused multiply-add (`a × b + c`) for fixed-point numbers with single-rounding behavior.
  * Supports both saturation (clamping) and modulo (wraparound) overflow modes.
  * Exposed via the fixed-point math library interface.

* **Tests**
  * Added a dedicated regression test suite validating correctness against an exact reference, including rounding behavior, overflow/corner cases, and identity properties.
  * Includes randomized and exhaustive checks across small fixed-point configurations.

### CODE REVIEW

File: include/sw/universal/number/fixpnt/math/fma.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+// wide fixed-point frame and added exactly, and only THEN is the result rounded
+// (round-to-nearest-even at bit rbits) and range-limited (Modulo wrap / Saturate
+// clamp). This matches fixpnt::operator*='s rounding model exactly, but with c
+// folded in before the single rounding -- so fma is strictly more accurate than
+// the two-rounding form fixpnt(a*b) + c whenever the product's low-order bits
+// matter (e.g. under cancellation, c ~ -a*b).
+//
+// Sub-issue of #1189 (universal fma). Relates to #1191.
+
+#include <universal/number/fixpnt/fixpnt_fwd.hpp>
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply include/sw/universal/number/fixpnt/math/fma.hpp:20]: Thanks -- I looked at this against the repo-wide convention and am keeping the current include on purpose.

Every number system's math partials (`cfloat/math/*.hpp`, `posit/math/*.hpp`, `lns/math/*.hpp`, and the fixpnt siblings `pow.hpp`/`hypot.hpp`/`exponent.hpp`/`truncate.hpp`) include *nothing* -- they are deliberately order-dependent partials that are only ever consumed through `mathlib.hpp`, which is included by the `fixpnt.hpp` umbrella *after* `fixpnt_impl.hpp`. So `blockbinary`, `urmul2`, `SpecificValue`, `BinaryNumberType`, `Modulo`, and `Saturate` are all in scope at the point `mathlib.hpp` pulls this file in.

`fma.hpp` is in fact already *more* self-contained than any sibling -- it carries a `#include <.../fixpnt_fwd.hpp>` that the others don't. Making it `#include fixpnt_impl.hpp` would make it the single outlier in a consistent, intentional pattern (mathlib orchestrates include order for the whole type). If the maintainer wants self-contained math partials, that's a library-wide change worth doing uniformly, not just for this one file.

Verified green building the `fixpnt_fma` target on both gcc and clang through the normal `fixpnt.hpp` include path.
